Sources/Amplitude/Plugins/iOS/UIKitUserInteractions.swift

```diff
@@ -0,0 +1,189 @@
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+import UIKit
+
+class UIKitUserInteractions {
+    fileprivate static let amplitudeInstances = NSHashTable<Amplitude>.weakObjects()
+
+    private static let lock = NSLock()
+
+    private static let initializeSwizzle: () = {
+        swizzleSendAction()
+    }()
+
+    private static let initializeNotificationListeners: () = {
+        NotificationCenter.default.addObserver(UITextField.self, selector: #selector(UITextField.amp_textFieldBeganEditing), name: UITextField.textDidBeginEditingNotification, object: nil)
+        NotificationCenter.default.addObserver(UITextField.self, selector: #selector(UITextField.amp_textFieldEndedEditing), name: UITextField.textDidEndEditingNotification, object: nil)
+    }()
+
+    static func register(_ amplitude: Amplitude) {
+        lock.withLock {
+            amplitudeInstances.add(amplitude)
+        }
+        initializeSwizzle
+        initializeNotificationListeners
+    }
+
+    private static func swizzleSendAction() {
+        let applicationClass = UIApplication.self
+
+        let originalSelector = #selector(UIApplication.sendAction)
+        let swizzledSelector = #selector(UIApplication.amp_sendAction)
+
+        guard
+            let originalMethod = class_getInstanceMethod(applicationClass, originalSelector),
+            let swizzledMethod = class_getInstanceMethod(applicationClass, swizzledSelector)
+        else { return }
+
+        let originalImp = method_getImplementation(originalMethod)
+        let swizzledImp = method_getImplementation(swizzledMethod)
+
+        class_replaceMethod(applicationClass,
+            swizzledSelector,
+            originalImp,
+            method_getTypeEncoding(originalMethod))
+        class_replaceMethod(applicationClass,
+            originalSelector,
+            swizzledImp,
+            method_getTypeEncoding(swizzledMethod))
+    }
+}
+
+extension UIApplication {
+    @objc func amp_sendAction(_ action: Selector, to target: Any?, from sender: Any?, for event: UIEvent?) -> Bool {
+        let sendActionResult = amp_sendAction(action, to: target, from: sender, for: event)
+
+        guard
+            sendActionResult,
+            let view = sender as? UIView
+        else { return sendActionResult }
+
+        if let textField = view as? UITextField, !textField.shouldTrack(action, for: event) {
+            return sendActionResult
+        } else {
+            #if !os(tvOS)
+            if let slider = view as? UISlider, !slider.shouldTrack(action, for: event) {
+                return sendActionResult
+            }
+            #endif
+        }
```

Comment:
Since everything here is trackable, then this just becomes:
```
if let trackable = view as? Trackable, !trackable.shouldTrack() {
  return sendActionResult
}
```

`Title` also seems like a good candidate to add to this protocol.